circuit/src/gadgets/select.rs

```diff
@@ -53,11 +61,23 @@ impl<F: FieldExt> SelectChip<F> {
 			let s_exp = v_cells.query_selector(s);
 
 			vec![
-				// bit * (a - b) - (r - b)
-				// a = 2
-				// b = 1
-				// r = 2
-				// 1 * (a - b) - (r - b) = 1 * 1 - (2 - 1) = 0
+				// bit * (x - y) - (z - y)
+				// z is the next rotation cell for the x value.
+				// Next rotation cell for x will carry the same value. (x == z)
+				// Example 1:
+				// bit = 1
+				// x = 5
+				// y = 3
+				// z = 5
+				// 1 * (x - y) - (z - y) = 1 * (5 - 3) - (5 - 3) = 0
+				// Example 2:
+				// bit = 0
+				// x = 5
+				// y = 3
+				// z = 5
+				// 0 * (x - y) - (z - y) = 0 * (5 - 3) - (5 - 3) = 2
```

Comment:
This does not evaluate to 0 because of the assumptions made above

Reply:
I think you meant it does evaluate to 0. I see now that our z value is also the selected result value here. It makes constraint == 0 and returns the selected variable.